src/utils/styled_lines_iterator.rs

```diff
@@ -0,0 +1,224 @@
+use utils::LinesIterator;
+use style::*;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Generates rows of styled text in constrained width.
+///
+/// Given styled text and a width constraint, it iterates over
+/// collections of styled spans, each making up a row within the constraint.
+///
+/// # Examples
+///
+/// ```
+/// # use cursive::style::{StyledString, StyledText};
+/// # use cursive::utils::StyledLinesIterator;
+/// # let text_styled: &[StyledString] = &[].as_ref();
+/// let text_plain = text_styled.to_plain();
+/// let spans = StyledLinesIterator::make_spans(text_styled);
+/// // You can now keep `text_plain` and `spans` alive while using `iter`.
```

Comment:
While updating `LinesIterator`, I found one of the reasons I was using indices: when inserting text in a `TextArea`, it tries not to recompute rows for the entire text, and just "offset" the indices in most situations.

Now, this is mostly an optimisation (so removing it should be slower, but safe), and `TextArea` may end up using an entirely different structure anyway.